torch/distributed/fsdp/flat_param.py

```diff
@@ -2214,20 +2214,31 @@ def _writeback_tensor(
             assert self.flat_param._is_grad_none_mask is not None
             self.flat_param._is_grad_none_mask[tensor_index] = True
 
-    def _clear_grads_if_needed(self):
-        """
-        When ``use_orig_params=True``, sets the underlying ``flat_param.grad``
-        to ``None`` if *all* of the original parameters' ``.grad`` are
-        ``None``. This is targeting ``optim.zero_grad(set_to_none=True)``, in
+    def _reset_flat_param_grad_info_if_needed(self):
+        """
+        When ``use_orig_params=True``:
+        (1) sets the underlying ``flat_param.grad`` to ``None`` if *all* of the
```

Comment:
Does this mean we need a unittest to ensure the flat_param.grad is None appropriately?

Reply:
This is the existing behavior, which we already have unit tests for in `test_fsdp_use_orig_params.py`.